os/arch/arm/src/lm3s6965-ek/src/up_boot.c

```diff
@@ -107,3 +112,86 @@ void tiva_boardinitialize(void)
 	board_led_initialize();
 #endif
 }
+
+
+#ifdef CONFIG_BOARD_INITIALIZE
+
+#if defined(CONFIG_QEMU_SRAM) || defined (CONFIG_QEMU_SDRAM)
+
+#define 	QEMU_SMARTFS_PARTITION_START	(0)
+#define 	QEMU_SMARTFS_PARTITION_SIZE	(1024 * 1024) /* 1MB */
+#define 	QEMU_SMARTFS_PARTITION_PARTNO	(0)
+#define 	QEMU_SMARTFS_PARTITION_MINORNO	(0)
+#define 	QEMU_SMARTFS_MOUNT_POINT	"/mnt"
+
+/************************************************************************************
+ * Name: board_initialize
+ *
+ * Description:
+ *	Board specific intialization after os app start
+ *
+ ************************************************************************************/
+
+int board_initialize(void)
+{
+	int ret;
+	char partref[4];
+	char devname[16];
+#ifdef CONFIG_MTD
+	FAR struct mtd_dev_s *mtd;
+	FAR struct mtd_geometry_s geo;
+	FAR struct mtd_dev_s *mtd_part;
+#endif
+	int partno = QEMU_SMARTFS_PARTITION_PARTNO;
+	int minorno = QEMU_SMARTFS_PARTITION_MINORNO;
+	int partoffset = QEMU_SMARTFS_PARTITION_START;
+	int partsize = QEMU_SMARTFS_PARTITION_SIZE;
+
+#ifdef CONFIG_MTD
+	mtd = up_flashinitialize();
+
+	if (!mtd) {
+		lldbg("ERROR : up_flashinitializ failed\n");
+		return 0;
+	}
+
+	if (mtd->ioctl(mtd, MTDIOC_GEOMETRY, (unsigned long)&geo) < 0) {
+		lldbg("ERROR: mtd->ioctl failed\n");
+		return 0;
+	}
+
+#ifdef CONFIG_MTD_PARTITION
+	mtd_part = mtd_partition(mtd, partoffset, partsize / geo.blocksize, partno);
+	if (!mtd_part) {
+		lldbg("ERROR: failed to create partition.\n");
+		return 0;
+	}
+#endif /* CONFIG_MTD_PARTITION */
+
+	snprintf(partref, sizeof(partref), "p%d", partno);
+	snprintf(devname, sizeof(devname), "/dev/smart%d%s", minorno, partref);
```

Comment:
Don't we need to check the return value of snprintf ?

Reply:
It is good to check